core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts

```diff
@@ -1239,22 +1287,36 @@ export class ConversationSkinComponent {
             this.explorationActuallyStarted = true;
           }
         }
-        if (!this.explorationPlayerStateService.isInQuestionMode()) {
+        if (this.explorationPlayerStateService.isInExplorationMode() ||
```

Comment:
I think you should have a new method that's something like isPresentingIsolatedQuestions() that returns true when the player is presenting questions, and false when it's in story. Document this method carefully to explain what the difference between true/false are, and use that method here.

In that method, do something like "if in these mode, return true, else if in these modes, return false, else throw exception". That way, if someone adds a new mode and forgets to update that, it gets caught.

That helps the reader understand the core concept of what we're basing this condition on, rather than "it's mode A or B or C".

Similarly above -- I think something like this will help clear some of the confusion and make the if conditions similar to understand. Also what is "other mode"? That just seems weird and probably needs to be communicated better -- the aim of the constructs I proposed is to avoid having a "miscellaneous category" because no one will have any idea what goes in it.

Reply:
Created the method.

For the ”other“ category, I have not seen any practical use case for it. 
Currently, it is only active while initializing the exploration player state service until any mode is set to the service.
I have removed the logic from the submit method to use “other mode”.